key-share/src/lib.rs

```diff
@@ -419,3 +426,94 @@ impl<T> From<ValidateError<T, InvalidCoreShare>> for InvalidCoreShare {
         err.into_error()
     }
 }
+
+/// Reconstructs a secret key from set of at least
+/// [`min_signers`](KeyShare::min_signers) key shares
+///
+/// Requires at least [`min_signers`](KeyShare::min_signers) distinct key shares
+/// from the same generation (key refresh produces key shares of the next
+/// generation). Accepts both [`KeyShare`] and [`IncompleteKeyShare`].
+/// Returns error if input is invalid.
+///
+/// Note that, normally, secret key is not supposed to be reconstructed, and key
+/// shares should never be at one place. This basically defeats purpose of MPC and
+/// creates single point of failure/trust.
+#[cfg(feature = "spof")]
+pub fn reconstruct_secret_key<E: Curve>(
+    key_shares: &[impl AsRef<CoreKeyShare<E>>],
+) -> Result<SecretScalar<E>, ReconstructError> {
```

Comment:
Doc is outdated. There's no "`KeyShare` and `IncompleteKeyShare`" in the `key_share` crate

Reply:
Can you add a CI job so it catches that stuff? This is an example from generic-ec:

https://github.com/dfns/generic-ec/blob/ba27154f93ec2502f71063056e1da35039140a69/.github/workflows/rust.yml#L81-L90

Reply:
Added CI, it required fixing some unrelated docs as well